### ISSUE TITLE
[MNG-6384] Project URL should not be extended with artifactId when inherited

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7485]
**[Richard Eckart de Castilho](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rec)** opened **[MNG-6384](https://issues.apache.org/jira/browse/MNG-6384?redirect=false)** and commented

According to the POM reference, the "url" element is informational and a "nice gesture" to users (https://maven.apache.org/pom.html ):

> name: Projects tend to have conversational names, beyond the artifactId. The Sun engineers did not refer to their project as "java-1.5", but rather just called it "Tiger". Here is where to set that value.
> description: Description of a project is always good. Although this should not replace formal documentation, a quick comment to any readers of the POM is always helpful.
> url: The URL, like the name, is not required. This is a nice gesture for projects users, however, so that they know where the project lives.
> inceptionYear: This is another good documentation point. It will at least help you remember where you have spent the last few years of your life.

However, Maven chooses to append the artifact ID when the "url" is inherited. IMHO that does not make any sense. In particular it does not make any sense to have to re-declare the project URL in **every child module** of a complex project which is usually inheriting from some project-specific top-level POM. It should be sufficient to declare the URL in the project's top-level POM and just have the children inherit it.

An inheriting project can usually not be expected to "live" under "url/artifactId". If the project truly lives somewhere else, then the POM should explicitly overwrite the URL. Otherwise, it should IMHO be assumed that the project continues to live in the same place.

---

**Issue Links:**
- [MNG-6059](https://issues.apache.org/jira/browse/MNG-6059) Important use cases not covered, as child.inherit.append.path affects all children
 (_**"is fixed by"**_)

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

for cases where such addition is not desired, a solution has been implemented in MNG-6059 for Maven 3.6.1

```xml
<project child.project.url.inherit.append.path="false">
  <url>...</url>
</project>
```

**Comment by jira-importer** [comment]
**[Richard Eckart de Castilho](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rec)** commented

Sounds like the new setting should resolve the issue described. However, looks like I cannot close the issue despite having created it.
